torch/csrc/lazy/backend/backend_interface.h

```diff
@@ -41,7 +42,7 @@ class TORCH_API BackendImplInterface {
 
   virtual const IrBuilder* GetIrBuilder() const = 0;
 
-  virtual bool ShouldSyncTensor(const LazyTensorPtr tensor) const;
+  virtual LazyGraphExecutor* GetLazyGraphExecutor() const;
```

Comment:
Since this is an API now, maybe a few words to describe it would be great.

Reply:
Also, have you compare this approach with providing a registration method directly in the LazyGraphExecutor class?

Reply:
> Also, have you compare this approach with providing a registration method directly in the LazyGraphExecutor class?

I have not. Do we anticipate the additional layer of indirection to have much impact on performance?

Reply:
I don't think performance is the main concern. Since the LazyGraphExecutor becomes a backend interface anyway, it just doesn't feel necessary to have the registration part in the BackendInterface to make the design cleaner.

Reply:
>  it just doesn't feel necessary to have the registration part in the BackendInterface to make the design cleaner

I'm not sure what you mean by this. When you say registration, are you referring the to static object initialization?

Reply:
I'm talking about things similar to `BackendRegistrar`.

Reply:
ah, I see now what you mean. I can implement that for `LazyGraphExecutor`